Update instructions to add new users/admins (#207)
scientific-python/specs: spec-0004/index.md

#### Process for Adding New Projects

After someone creates an issue on https://github.com/scientific-python/upload-nightly-action
requesting access to upload wheels a human/admin has to respond to that request.

Admins are people from the community who are ideally not part of the same organizations nor
project. This is to prevent malicious activities from a given group of actors and ensure a
diverse and healthy community. Adding new people to the list of admins requires at
least an issue to be openned.

We want to be open to projects uploading wheels but at the same time need to perform some
amount of due dilligence before giving people access. This is because once a user is given
access their work will be broadcasted through the broad exposure of Scientific Python. This
could be abused to publish malicious packages.

Once you have established who the person is and that they represent the project they want
to upload wheels for ask the person to create an account on https://anaconda.org and tell

> you the username. We suggest that projects have at least 2 representatives. When considering representatives, we suggest that projects nominate individuals that do not have significant community, organizational, or employer overlap with existing representatives to ensure that we have a diverse community.

Author: ```suggestion
you the username. We suggest that projects have at least 2 representatives.
When considering representatives, we suggest that projects nominate individuals
that do not have significant community, organizational, or employer overlap
with existing representatives to ensure that we have a diverse community.
```

#### Process for Adding New Projects

After someone creates an issue on https://github.com/scientific-python/upload-nightly-action
requesting access to upload wheels a human/admin has to respond to that request.

Admins are people from the community who are ideally not part of the same organizations nor
project. This is to prevent malicious activities from a given group of actors and ensure a

> diverse and healthy community. Adding new people to the list requires of admins requires at

Reviewer: ```suggestion
diverse and healthy community. Adding new people to the list of admins requires at
```

least an issue to be openned.

We want to be open to projects uploading wheels but at the same time need to perform some
amount of due dilligence before giving people access. This is because once a user is given

> access they benefit the wide exposure of Scientific Python and could take this to their
> advantage to publish malicious packages.

Reviewer 2: ```suggestion
access their work will be broadcasted through the broad exposure of Scientific Python. This
could be abused to publish malicious packages. 
```

Reviewer 2: I think this was the essence of what you were trying to say. I rephrased for clarity but am happy to hear alternative suggestions.

Author: Yes thanks!

Once you have established who the person is and that they represent the project they want
to upload wheels for ask the person to create an account on https://anaconda.org and tell

> you the username. We suggest projects to have a least 2 representatives and ideally they

Reviewer 2: ```suggestion
you the username. We suggest that projects have a least 2 representatives and ideally they
```

> should not be part of the same organization to ensure a diverse community.

Reviewer 2: what do we mean by organization here? I think this is worth clarifying, because a project could also be interpreted as an organization?

Author: I mean like you belong to the same project, company or OSS community.

Reviewer 2: Ok, so the sentence before this says that projects have at least 2 representatives, but that they should not be part of the same project, company, or OSS community? Maybe we could say something like 

> We suggest that projects have at least 2 representatives. When considering representatives, we suggest that projects nominate individuals that do not have significant community, organizational, or employer overlap with existing representatives to ensure that we have a diverse community.

Author: I added your suggestion thanks!

> you the username. We suggest that projects have a least 2 representatives and ideally they
> should not be part of the same organization to ensure a diverse community.

Author: ```suggestion
you the username. We suggest that projects have at least 2 representatives. When considering representatives, we suggest that projects nominate individuals that do not have significant community, organizational, or employer overlap with existing representatives to ensure that we have a diverse community.
```

You (as an admin) can then generate a personal access token at
https://anaconda.org/[user]/settings/access.
The token should only have the "Allow uploads to Standard Python repositories" and
"Allow write access to the API site" scope. The creation of tokens at the organization
level should be avoided for security reasons.

> Then you need to do a first upload of a wheel to create the package listing on anaconda.org.

Reviewer 2: Do you mean that an existing admin uploads the wheel for the new admin? Or is the "you" here switching from the previous paragraph?

Author: Yeah this part is annoying. Here I was suggesting that an admin of scientific-python would do the first upload, otherwise you can communicate the token to the new person. Which is an issue as you would communicate an admin token to someone else and this person could either do a rogue action or if not it's just an admin token in the nature and you really must not forget to revoke the token once the package is up.

Reviewer 2: Ahhh ok! If the admin is still referring to the same person then I have no issue with this phrasing.
